okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -494,13 +494,10 @@ public boolean isEligible(Address address, @Nullable Route route) {
     // 2. The routes must share an IP address. This requires us to have a DNS address for both
     // hosts, which only happens after route planning. We can't coalesce connections that use a
     // proxy, since proxies don't tell us the origin server's IP address.
-    if (route == null) return false;
-    if (route.proxy().type() != Proxy.Type.DIRECT) return false;
-    if (this.route.proxy().type() != Proxy.Type.DIRECT) return false;
-    if (!this.route.socketAddress().equals(route.socketAddress())) return false;
+    if (routes == null || !routeMatchesAny(routes)) return false;
```

Comment:
The comment above can probably move to the method, below.

Reply:
Can fix in follow-up!